public/app/features/alerting/unified/components/receivers/form/ChannelOptions.tsx

```diff
@@ -29,14 +29,15 @@ export function ChannelOptions<R extends ChannelValues>({
   readOnly = false,
 }: Props<R>): JSX.Element {
   const { watch } = useFormContext<ReceiverFormValues<R>>();
-  const currentFormValues = watch() as Record<string, any>; // react hook form types ARE LYING!
+  const currentFormValues = watch(); // react hook form types ARE LYING!
   return (
     <>
       {selectedChannelOptions.map((option: NotificationChannelOption, index: number) => {
         const key = `${option.label}-${index}`;
         // Some options can be dependent on other options, this determines what is selected in the dependency options
         // I think this needs more thought.
-        const selectedOptionValue = currentFormValues[`${pathPrefix}settings.${option.showWhen.field}`];
+        const path = `${pathPrefix}settings.${option.showWhen.field}`.split('.');
+        const selectedOptionValue = path.reduce((o, n) => o[n], currentFormValues) as string;
```

Comment:
This is generating a type error at the moment (see https://drone.grafana.net/grafana/grafana/79196/1/6) , and because you didn't run the precommit hook it also fails our linter (https://drone.grafana.net/grafana/grafana/79196/1/7) which would have preventing you from committing this in the first place.

Reply:
i resubmitted